homework4/task1_round_nums.py

```diff
@@ -0,0 +1,13 @@
+"""Программа, которая вычисляет число c заданной точностью d"""
+
+from decimal import Decimal
+
+number = Decimal(input('Введите число: '))
+
+while True:
+    quantity = input('Введите точность округления d в диапазоне [10^(-1) ≤ d ≤10^(-10)]: ')
+    if 1 / float(quantity) % 10 == 0 and quantity.count('0') in range(1,11):
+        break
+
+output_number = number.quantize(Decimal("1." + '0' * quantity.count('0')))
```

Comment:
Зачем такие сложности
```suggestion
output_number = number.quantize(Decimal(quantity))
```